pkg/audit/containment.go

```diff
@@ -0,0 +1,71 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package audit
+
+import (
+	"context"
+
+	"go.uber.org/zap"
+
+	"storj.io/storj/pkg/pb"
+	"storj.io/storj/pkg/storj"
+)
+
+// PendingAudit contains info needed for retrying an audit for a contained node
+type PendingAudit struct {
+	NodeID            storj.NodeID
+	PieceID           storj.PieceID
+	StripeIndex       int
+	ShareSize         int64
+	ExpectedShareHash []byte
+	ReverifyCount     int
+}
+
+// DB holds information about pending audits for contained nodes
+type DB interface {
+	Get(ctx context.Context, nodeID pb.NodeID) (*PendingAudit, error)
+	IncrementPending(ctx context.Context, pendingAudit *PendingAudit) error
+	Delete(ctx context.Context, nodeID pb.NodeID) error
+}
+
+// Containment allows access to pending audit info in the db
+type Containment struct {
+	log *zap.Logger
+	db  DB
+}
+
+// NewContainment creates a new containment db
+func NewContainment(log *zap.Logger, db DB) *Containment {
+	return &Containment{
+		log: log,
+		db:  db,
+	}
+}
+
+// Get will get pending audit info from the db
+func (containment *Containment) Get(ctx context.Context, nodeID pb.NodeID) (*PendingAudit, error) {
+	pendingAudit, err := containment.db.Get(ctx, nodeID)
+	if err != nil {
+		return nil, ContainError.Wrap(err)
+	}
+	return pendingAudit, nil
+}
+
+// IncrementPending will either create a new entry for a pending audit or update an existing pending audit's reverify count
+func (containment *Containment) IncrementPending(ctx context.Context, pending *PendingAudit) error {
+	err := containment.db.IncrementPending(ctx, pending)
+	if err != nil {
+		return ContainError.Wrap(err)
+	}
+	return nil
+}
+
+// Delete will delete pending audit info from the db
+func (containment *Containment) Delete(ctx context.Context, nodeID pb.NodeID) error {
+	err := containment.db.Delete(ctx, nodeID)
+	if err != nil {
+		return ContainError.Wrap(err)
+	}
+	return nil
+}
```

Comment:
Shouldn't the Delete function also set the contained flag for the node to false? Maybe I'm just missing where it's done?

Reply:
that's a good point! it does need to happen here I forgot to add it

Reply:
okay actually I think I'm going to add that in a separate PR, because it looks like there's not any functions on the overlay cache that will allow me to update the Contained field right now (UpdateNodeInfo only allows updating "info requested from the node itself like node type, email, wallet, capacity, and version").

Reply:
My previous question about the usage of this wrapper also came because it's currently a wrapper which doesn't provided anything on top of the _containment_ struct, however with this addition is justifying it aside of your response to my question.

Reply:
opened a pr to add this functionality: https://github.com/storj/storj/pull/2013